src/main/java/org/scijava/module/DefaultModuleService.java

```diff
@@ -478,24 +474,25 @@ private void assignInputs(final Module module,
 	}
 
 	private <T> ModuleItem<T> getTypedSingleItem(final Module module,
-		final Class<T> type, final Iterable<ModuleItem<?>> items)
+		final Class<T> type, final Iterable<ModuleItem<?>> items, boolean acrossTypes)
 	{
 		Set<Class<?>> types = new HashSet<>();
 		types.add(type);
 		@SuppressWarnings("unchecked")
-		ModuleItem<T> result = (ModuleItem<T>) getSingleItem(module, types, items);
+		ModuleItem<T> result = (ModuleItem<T>) getSingleItem(module, types, items, acrossTypes);
 		return result;
 	}
 
 	private ModuleItem<?> getSingleItem(final Module module,
-		final Collection<Class<?>> types, final Iterable<ModuleItem<?>> items)
+		final Collection<Class<?>> types, final Iterable<ModuleItem<?>> items, boolean acrossTypes)
 	{
 		ModuleItem<?> result = null;
 
 		for (final ModuleItem<?> item : items) {
 			final String name = item.getName();
 			if (!item.isAutoFill()) continue; // skip unfillable inputs
 			if (module.isInputResolved(name)) continue; // skip resolved inputs
```

Comment:
While working on this pull request, I noticed that the above line is specific to *inputs* only, so the method probably doesn't show the intended behavior when called with *outputs*.

The current unit tests seem to miss the method signatures specific to outputs (`getSingleOutput`).

I'll open a separate issue for this.